cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java

```diff
@@ -112,31 +123,92 @@ public MetricsHandler(Authenticator authenticator,
     this.metricStore = metricStore;
   }
 
+  // Deprecate search with context param
   @POST
   @Path("/search")
   public void search(HttpRequest request, HttpResponder responder,
                      @QueryParam("target") String target,
-                     @QueryParam("context") String context) throws IOException {
+                     @QueryParam("context") String context,
+                     @QueryParam("tag") List<String> tags) throws IOException {
     if (target == null) {
       responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Required target param is missing");
       return;
     }
 
-    if ("childContext".equals(target)) {
+    if (target.equals("tag")) {
+      searchTargetContextAndRespond(responder, tags);
```

Comment:
what does searchTargetContext mean? Not clear name for the guy reading the code...